net/SslSocket.hpp

```diff
@@ -147,6 +147,28 @@ class SslStreamSocket final : public StreamSocket
                       int64_t & timeoutMaxMicroS) override
     {
         ASSERT_CORRECT_SOCKET_THREAD(this);
+
+        // We cannot communicate with a client when SSL wants
+        // to negotiate the handshake. Here, we give priority to
+        // SSL's needs. Only when SSL is done negotiating
+        // (i.e. wants neither to read, nor to write) do we see
+        // what data activity we have.
+        if (_doHandshake)
```

Comment:
can we put a #if TESTING type thing around this - so that we isolate the functional change only for the 22.05 branch ? - we should have a configure switch that sets that and ask Andras to have that on for CODE (I guess) ? =)